service-python/main_consumer.py

```diff
@@ -44,17 +45,25 @@ def setup_queues(self):
         self.channel = channel
 
 
+# This file will get copied to the docker image's root folder(src) when being built
+# When run, it attempts to create a pika.BlockingConnection() with the settings in CONSUMER_CONFIG
+# There are 2 retry levels for the consumer. The first being in _create_connection() which is based on retry_limit
+# The other being when the app crashes which is based on timeout
+# If it fails, it will try to delete any existing connection and the reference to the instantiated class
+# since the retry loop will recreate it
 if __name__ == "__main__":
 
     start_timer = None
     current_timer = None
 
     while(True):
+        consumer = None
         try:
             consumer = RabbitMQConsumer(CONSUMER_CONFIG)
```

Comment:
This I do not understand. Suppose _create_connection failed in all tries (so self.connection is None)? What then? the consumer will not be None and we will happily call start_consuming?